Sources/NIO/Thread.swift

```diff
@@ -57,31 +80,46 @@ final class NIOThread {
     /// - parameters:
     ///     - body: The closure that will accept the `pthread_t`.
     /// - returns: The value returned by `body`.
-    func withUnsafePthread<T>(_ body: (pthread_t) throws -> T) rethrows -> T {
-        return try body(self.pthread)
+#if !os(Windows)
+    internal func withUnsafePthread<T>(_ body: (pthread_t) throws -> T) rethrows -> T {
+        return try body(self.handle)
     }
+#endif
 
     /// Get current name of the `NIOThread` or `nil` if not set.
     var currentName: String? {
-        get {
-            // 64 bytes should be good enough as on Linux the limit is usually 16 and it's very unlikely a user will ever set something longer anyway.
-            var chars: [CChar] = Array(repeating: 0, count: 64)
-            return chars.withUnsafeMutableBufferPointer { ptr in
-                guard sys_pthread_getname_np(self.pthread, ptr.baseAddress!, ptr.count) == 0 else {
-                    return nil
-                }
-
-                return String(decoding: UnsafeRawBufferPointer(UnsafeBufferPointer<CChar>(rebasing: ptr.prefix { char in
-                                  char != 0
-                              })),
-                              as: Unicode.UTF8.self)
+#if os(Windows)
+        var pszBuffer: PWSTR?
+        GetThreadDescription(self.handle, &pszBuffer)
+        guard let buffer = pszBuffer else { return nil }
+        let string: String = String(decodingCString: buffer, as: UTF16.self)
```

Comment:
don't we need `Unicode.UTF16.self` here? I thought with Swift 5 you always need the `Unicode.` prefix?

Reply:
Hmm, I've never needed the `Unicode` prefix.  I'm happy to add it if thats the style that the project is using, but this builds fine (and a similar pattern is used extensively in Foundation).

Reply:
ha, maybe they used to issue a warning but saw too many breakages and removed it again :) . happy if it doesn't warn today .